neonvm/apis/neonvm/v1/virtualmachine_webhook.go

```diff
@@ -172,6 +187,26 @@ func (r *VirtualMachine) ValidateUpdate(old runtime.Object) error {
 		}
 	}
 
+	// validate swap changes by comparing the SwapInfo for each.
+	//
+	// If there's an error with the old object, but NOT an error with the new one, we'll allow the
+	// new one to proceed. This is to prevent any VirtualMachine objects getting stuck during
```

Comment:
Can you clarify why the objects would get stuck if this logic wouldn't exist?

Reply:
Done — tl;dr the webhook can reject potential object updates. If we refuse to accept changes because the _old_ state is bad, then it isn't possible to fix any objects if they end up broken.